src/dosext/mfs/mfs.c

```diff
@@ -716,27 +716,73 @@ int dos_utime(char *fpath, struct utimbuf *ut)
   return -1;
 }
 
-int dos_get_disk_space(const char *cwd, unsigned int *free, unsigned int *total,
-		       unsigned int *spc, unsigned int *bps)
+static int dos_get_disk_space(const char *cwd, unsigned int *dfree,
+                              unsigned int *total, unsigned int *spc,
+                              unsigned int *bps)
 {
   struct statfs fsbuf;
 
-  if (statfs(cwd, &fsbuf) >= 0) {
-    /* return unit = 512-byte blocks @ 1 spc, std for floppy */
-    *spc = 1;
-    *bps = 512;
-    *free = (fsbuf.f_bsize / 512) * fsbuf.f_bavail;
-    *total = (fsbuf.f_bsize / 512) * fsbuf.f_blocks;
-
-    while (*spc < 64 && *total > 65535) {
```

Comment:
This is a wrong thing to do.
See here:
https://support.microsoft.com/en-us/help/140365/default-cluster-size-for-ntfs--fat--and-exfat
for cluster sizes depending on a volume size.
Now for FAT32 API you did spc==1 always,
disregarding the volume size. I wouldn't be
surprised if this causes the error in some DOSes.